backend/prompt_studio/prompt_studio/views.py

```diff
@@ -41,3 +47,94 @@ def get_queryset(self) -> Optional[QuerySet]:
+        prompts to reflect the new order. If the start and end sequence numbers
+        are equal, it returns a bad request response.
+
+        Args:
+            request (HttpRequest): The HTTP request object containing the data to
+            reorder prompts.
+
+        Returns:
+            Response: A Response object with the status of the reordering operation.
+        """
+        try:
+            # Validate request data
+            serializer = ReorderPromptsSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+
+            # Extract validated data from the serializer
+            start_sequence_number = serializer.validated_data.get(
+                ToolStudioPromptKeys.START_SEQUENCE_NUMBER
+            )
+            end_sequence_number = serializer.validated_data.get(
+                ToolStudioPromptKeys.END_SEQUENCE_NUMBER
+            )
+            prompt_id = serializer.validated_data.get(ToolStudioPromptKeys.PROMPT_ID)
+
+            prompt_instance: ToolStudioPrompt = ToolStudioPrompt.objects.get(
+                pk=prompt_id
+            )
+            filtered_prompts_data = []
+            tool_id = prompt_instance.tool_id
+
+            # Determine the direction of sequence adjustment based on start and
+            # end sequence numbers
+            if start_sequence_number < end_sequence_number:
+                logger.info(
+                    "Start sequence number is less than end sequence number. "
+                    "Decrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__gt": start_sequence_number,
+                    "sequence_number__lte": end_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
+                    filters, increment=False
+                )
+
+            elif start_sequence_number > end_sequence_number:
+                logger.info(
+                    "Start sequence number is greater than end sequence number. "
+                    "Incrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__lt": start_sequence_number,
+                    "sequence_number__gte": end_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
+                    filters, increment=True
+                )
+
+            # Update the sequence number of the moved prompt
+            prompt_instance.sequence_number = end_sequence_number
+            prompt_instance.save()
```

Comment:
@tahierhussain Can we move the core logic and DB calls to a helper class? It is better to have service layer in the views.

Reply:
@harini-venkataraman Done